keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,404 @@
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The feature can be disabled in Alpha and Beta versions by restarting kube-controller-manager, kube-scheduler and kube-apiserver with the feature-gate off. 
+In terms of Stable versions, users can choose to opt-out by not setting the corresponding field.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The `RequiredDuringSchedulingRequiredDuringExecution` node affinity will take effect as expected.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The change is opt-in, it doesn't impact already running workloads. 
+
+###### What specific metrics should inform a rollback?
+
+Each time the NodeAffinityController evicts a pod, it will add an event with reason "NodeLabelsManagerEviction" to the cluster's events. When user see lots of this type of events, they should pay attention to this feature.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Yes.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Inspect the nodeAffinity.RequiredDuringSchedulingRequiredDuringExecution configuration
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: NodeLabelsManagerEviction
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [x] Other (treat as last resort)
+  - Details: pod with `RequiredDuringSchedulingRequiredDuringExecution` must meet affinity rule during both the scheduling phase and the execution phase.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: 
+    - pod_deletions_total(counts the number of Pods deleted since its start)
+    - pod_deletion_duration_seconds(tracks the latency between the time when a node affinity effect has been activated for the Pod and its deletion.)
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
```

Comment:
Probably worth a metric counter to count the evictions.

Reply:
Mentioned two new metrics in this KEP.
```
  - Metric name:
    - pod_deletions_node_affinity_total(counts the number of Pods deleted because of nodeAffinity since this contoller start)
    - pod_deletions_node_affinity_duration_seconds(Latency, in seconds, between the time when a node's label has been changed and the pod been deleted via AffinityEvictionController)
```
I think this is enough?